### PR TITLE
refactor!: Remove explicit geometry ID from source link

### PR DESCRIPTION
**Pull request by paulgessinger**
This reduces the memory footprint. The access to the surface is now done using a delegate (which I typedef right next to Source Link itself). Components need to handle this individually.

### CODE REVIEW

File: Core/include/Acts/EventData/SourceLink.hpp
Comment (benjaminhuth) [inline Core/include/Acts/EventData/SourceLink.hpp:1]: Hmm the `m_geometryId` member seems to be present still... This should be removed to actually reduce the memory footprint, right?

**Review by benjaminhuth — COMMENTED** [review]
Only one thing I would like to clarify, otherwise looks good!

File: Core/include/Acts/EventData/SourceLink.hpp
Comment (benjaminhuth) [reply Core/include/Acts/EventData/SourceLink.hpp:1]: can you clarify on that @paulgessinger ?

Comment (paulgessinger) [reply Core/include/Acts/EventData/SourceLink.hpp:1]: Sorry, I missed this. Yes indeed that member needs to go!

**Comment by benjaminhuth** [comment]
Still some tests seem to rely on the old interface @paulgessinger

**Comment by paulgessinger** [comment]
This should hopefully finally be good.

**Comment by paulgessinger** [comment]
@benjaminhuth ?

**Review by benjaminhuth — APPROVED** [review]
Looks good!

**Comment by paulgessinger** [comment]
RTD is stuck, I'm merging manually.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31988630) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6102879)!

**Pull request merged**